GUI/src/main/kotlin/ui/components/general/ProjectManager.kt

```diff
@@ -98,11 +118,30 @@ fun handleSaveProject(
     path: String,
 ) {
     var savePath = path
-    if (!savePath.endsWith(".json")) {
-        savePath = "$savePath.json"
+    // add .mkv extension if not present
+    if (!savePath.endsWith(".mkv")) {
+        savePath = "$savePath.mkv"
     }
+    // set save path
     state.value.saveProjectPath = savePath
 
-    val jsonData = JsonMapper.mapper.writeValueAsString(state.value)
-    File(savePath).writeText(jsonData)
+    val grabber = FFmpegFrameGrabber(state.value.outputPath)
+    grabber.start()
+    val recorder = FFmpegFrameRecorder(savePath, grabber.imageWidth, grabber.imageHeight)
+    // set metadata
+    recorder.setMetadata("APP-STATE", JsonMapper.mapper.writeValueAsString(state.value))
+    recorder.start()
+    // clone
+    var frame: Frame? = grabber.grabFrame()
```

Comment:
Do we really have to grab and record each frame manually? Isnt there such a function as to just copy a video and change the metadata?

Reply:
https://stackoverflow.com/a/11479066. at least ffmpeg as cli offers this feature. But I dont know about javacv/ffmpeg...